common/rpc/peer_chooser.go

```diff
@@ -49,12 +84,32 @@ func NewDNSPeerChooserFactory(interval time.Duration, logger log.Logger) PeerCho
 	return &dnsPeerChooserFactory{interval, logger}
 }
 
-func (f *dnsPeerChooserFactory) CreatePeerChooser(transport peer.Transport, address string) (peer.Chooser, error) {
+func (f *dnsPeerChooserFactory) CreatePeerChooser(transport peer.Transport, opts PeerChooserOptions) (PeerChooser, error) {
 	peerList := roundrobin.New(transport)
-	peerListUpdater, err := newDNSUpdater(peerList, address, f.interval, f.logger)
+	peerListUpdater, err := newDNSUpdater(peerList, opts.Address, f.interval, f.logger)
 	if err != nil {
 		return nil, err
 	}
 	peerListUpdater.Start()
-	return peerList, nil
+	return &defaultPeerChooser{Chooser: peerList}, nil
```

Comment:
any reason for the wrapper instead of just returning `peerList`?  they appear to have the same API.

Reply:
Peer chooser factories need to return our new `PeerChooser` interface which has additional `UpdatePeers()` func.